.github/workflows/pr-tests.yml

```diff
@@ -65,9 +64,71 @@ jobs:
           ~/.cargo/git
           target
         key: ${{ runner.os }}-cargo-pr-tests-${{ hashFiles('**/Cargo.toml') }}
+
+  test_quick:
+    needs: build
```

Comment:
Should `test_quick` and `test_slow` also run after the merge to main?
If not, we could add `if: github.ref != 'refs/heads/main'` here and on the `test_slow` section.